File: Untitled-1.cpp
#include <iostream>
#include <string>
using namespace std;

struct Student {
    int id;
    string name;
    float score;
    Student* next;
};

int main() {
    Student* head = NULL;
    int n;

    cout << "Nhap so hoc sinh: ";
    cin >> n;
    cin.ignore();

    // Nhập dữ liệu
    for(int i = 0; i < n; i++) {
        Student* sv = new Student;
        cout << "\nHoc sinh " << i+1 << ":\n";
        cout << "ID: "; cin >> sv->id; cin.ignore();
        cout << "Ten: "; getline(cin, sv->name);
        cout << "Diem: "; cin >> sv->score; cin.ignore();
        sv->next = NULL;

        // Thêm vào danh sách
        if(!head) head = sv;
        else {
            Student* t = head;
            while(t->next) t = t->next;
            t->next = sv;
        }
    }

    // Sắp xếp điểm giảm dần 
    for(Student* i = head; i; i = i->next) {
        for(Student* j = i->next; j; j = j->next) {
            if(i->score < j->score) {
                swap(i->id, j->id);
                swap(i->name, j->name);
                swap(i->score, j->score);
            }
        }
    }

    // In kết quả
    cout << "\n===DIEM===\n";
    for(Student* p = head; p; p = p->next) {
        cout << p->id << " - " << p->name << " - " << p->score << endl;
    }

    return 0;
}

File: Untitled-1.java
import java.util.*;

class Student {
    int id;
    String name;
    float score;

    public Student(int id, String name, float score) {
        this.id = id;
        this.name = name;
        this.score = score;
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        ArrayList<Student> list = new ArrayList<>();

        System.out.print("Nhap so hoc sinh: ");
        int n = sc.nextInt();
        sc.nextLine();

        // Nhập dữ liệu
        for (int i = 0; i < n; i++) {
            System.out.println("\nHoc sinh " + (i + 1) + ":");

            System.out.print("ID: ");
            int id = sc.nextInt();
            sc.nextLine();

            System.out.print("Ten: ");
            String name = sc.nextLine();

            System.out.print("Diem: ");
            float score = sc.nextFloat();
            sc.nextLine();

            list.add(new Student(id, name, score));
        }

        // Sắp xếp giảm dần theo điểm
        Collections.sort(list, (a, b) -> Float.compare(b.score, a.score));

        // In kết quả
        System.out.println("\n===DIEM===");
        for (Student s : list) {
            System.out.println(s.id + " - " + s.name + " - " + s.score);
        }

        sc.close();
    }
}